File: frontend/src/types/news.ts
export interface EntityExposure {
  ticker?: string | null;
  sector?: string | null;
  entity_type: string;
  exposure_type: string;
  sentiment_hint: string;
}

export interface NewsItem {
  id: string;
  title: string;
  summary?: string | null;
  source: string;
  source_url?: string | null;
  published_at: string;
  fetched_at: string;
  category: string;
  subcategory?: string | null;
  importance: "HIGH" | "MEDIUM" | "LOW" | string;
  raw_content?: string | null;
  entities: EntityExposure[];
}

export interface NewsListResponse {
  total: number;
  news: NewsItem[];
}

export interface NewsMetadataResponse {
  sources: string[];
  categories: string[];
  total_count: number;
}

export interface NewsSyncResponse {
  status: string;
  inserted_count: number;
  message: string;
}

export interface NewsFilterParams {
  category?: string;
  importance?: string;
  source?: string;
  search?: string;
  ticker?: string;
  company?: string;
  limit?: number;
  offset?: number;
}

File: app/models/news.py
from dataclasses import dataclass, field
from typing import Optional, List
from enum import Enum


class NewsCategory(str, Enum):
    MACRO_ECONOMY = "Macro & Economy"
    POLICY_REGULATION = "Policy & Regulation"
    COMMODITIES_ENERGY = "Commodities & Energy"
    GEOPOLITICAL = "Geopolitical & Trade"
    SECTOR_INDUSTRY = "Sector & Industry"
    CORPORATE_ACTION = "Corporate & Earnings"
    GLOBAL_MARKETS = "Global Markets"
    GENERAL = "General Market"


class ExposureType(str, Enum):
    DIRECT = "DIRECT"
    UPSTREAM = "UPSTREAM"
    DOWNSTREAM = "DOWNSTREAM"
    SENSITIVE = "SENSITIVE"


class SentimentHint(str, Enum):
    POSITIVE = "POSITIVE"
    NEGATIVE = "NEGATIVE"
    MIXED = "MIXED"
    NEUTRAL = "NEUTRAL"


@dataclass
class EntityExposure:
    ticker: Optional[str]
    sector: Optional[str]
    entity_type: str  # 'COMPANY', 'SECTOR', 'COMMODITY', 'POLICY'
    exposure_type: ExposureType
    sentiment_hint: SentimentHint


@dataclass
class NewsItem:
    id: str  # SHA-256 hash of (source + title + published_at)
    title: str
    summary: Optional[str]
    source: str
    source_url: Optional[str]
    published_at: str
    fetched_at: str
    category: str
    subcategory: Optional[str] = None
    importance: str = "MEDIUM"  # 'HIGH', 'MEDIUM', 'LOW'
    raw_content: Optional[str] = None
    entities: List[EntityExposure] = field(default_factory=list)
